crypto/encode_test.go

```diff
@@ -15,15 +16,17 @@ type byter interface {
 	Bytes() []byte
 }
 
-func checkAminoBinary(t *testing.T, src byter, dst interface{}, size int) {
+func checkAminoBinary(t *testing.T, src, dst interface{}, size int) {
 	// Marshal to binary bytes.
 	bz, err := cdc.MarshalBinaryBare(src)
 	require.Nil(t, err, "%+v", err)
-	// Make sure this is compatible with current (Bytes()) encoding.
-	require.Equal(t, src.Bytes(), bz, "Amino binary vs Bytes() mismatch")
+	if byterSrc, ok := src.(byter); ok {
+		// Make sure this is compatible with current (Bytes()) encoding.
+		assert.Equal(t, byterSrc.Bytes(), bz, "Amino binary vs Bytes() mismatch")
+	}
 	// Make sure we have the expected length.
 	if size != -1 {
-		require.Equal(t, size, len(bz), "Amino binary size mismatch")
+		assert.Equal(t, size, len(bz), "Amino binary size mismatch")
```

Comment:
Why assert instead of require? Wouldn't we want this to fail immediately?

Reply:
I was changing this to match whats in tendermint/crypto right now. I don't really have a preference here.

Reply:
I think it's most important to be consistent -- it seems in SDK-land, the defacto call is `require`.

Reply:
Its to use require unless there is any reason not to. There is good reason to not use it here if we had more pubkey types, but I guess we can change it when that happens.